router/core/router.go

```diff
@@ -1850,13 +1854,15 @@ func DefaultTransportRequestOptions() *TransportRequestOptions {
 }
 
 func NewSubgraphTransportOptions(cfg config.TrafficShapingRules) *SubgraphTransportOptions {
+	allRequestOptions := NewTransportRequestOptions(cfg.All, nil)
```

Comment:
From how I understand this, we previously copied over the default value when the subgraph specific value exists, however now we apply the value from "all" (with the defaults applied to all first).

PR looks ok but I believe this could break backward compatiblity, if someone expects the default value to be applied, INSTEAD of the value from all.

Should we introduce a flag to enable this?

Reply:
I think the current behaviour is unintuitive enough that if we "break" this more users are likely to get expected results with the new version than unexpected ones.

An enabled-by-default flag would be semantically equivalent to just changing the config to have the value they want, and a disabled-by-default one would leave it "broken" for the vast majority of users.

Reply:
Fixing this is more important than ensuring default value behaviour. The inherent values should already be reasonable, if not the same